tests/tests_pdb.rs.in

```diff
@@ -4,21 +4,17 @@ fn pdb_{name}() {{
     println!("Parsing file: {filepath}");
     let data = include_bytes!("{filepath}").as_slice();
     let mut reader = Cursor::new(data);
-    let header = Header::read(&mut reader).expect("failed to parse header");
-    println!("{{:#?}}", header);
+    let mut db = Database::open_non_persistent(&mut reader, DatabaseType::Plain).expect("failed to open database");
+    println!("{{:#?}}", db.get_header());
 
-    for (i, table) in header.tables.iter().enumerate() {{
-        println!("Table {{}}: {{:?}}", i, table.page_type);
-        let pages = header.read_pages(
-                &mut reader,
-                binrw::Endian::NATIVE,
-                (&table.first_page, &table.last_page, DatabaseType::Plain),
-            )
-            .expect("failed to read pages");
+    let table_ids = db.get_header().tables.iter().enumerate().map(|(i, _)| i).collect::<Vec<_>>();
+    for i in table_ids {{
```

Comment:
Can we make this more ergonomic? I suppose we can't iter here because we need a mutable reference in the loop body, right?